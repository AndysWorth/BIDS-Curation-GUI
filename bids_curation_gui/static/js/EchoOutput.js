import React from 'react';
import PropTypes from 'prop-types';

/**
 * Alert component to render echo from the server.
 *
 * @param {String} value
 */
const EchoOutput = ({value}) => (
  <div className="alert alert-primary" hidden={value ? false : true}>
    <span className="badge badge-light">Server says:</span> {value}
  </div>
);

EchoOutput.propTypes = {
  value: PropTypes.string.isRequired,
};

export default EchoOutput;