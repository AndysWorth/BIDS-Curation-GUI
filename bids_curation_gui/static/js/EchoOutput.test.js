import React from 'react';
import { shallow } from 'enzyme';
import EchoOutput from './EchoOutput';

describe('EchoOutput', () => {
  it('should render an initial value', () => {
      const wrapper = shallow(<EchoOutput value="test-value" />);
      expect(wrapper).toMatchSnapshot();
  });

  it('should be hidden when there is a blank value', () => {
      const wrapper = shallow(<EchoOutput value="" />);
      expect(wrapper).toMatchSnapshot();
  });
});