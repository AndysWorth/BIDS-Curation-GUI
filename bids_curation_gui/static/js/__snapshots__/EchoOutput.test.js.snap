// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EchoOutput should be hidden when there is a blank value 1`] = `
<div
  className="alert alert-primary"
  hidden={true}
>
  <span
    className="badge badge-light"
  >
    Server says:
  </span>
   
</div>
`;

exports[`EchoOutput should render an initial value 1`] = `
<div
  className="alert alert-primary"
  hidden={false}
>
  <span
    className="badge badge-light"
  >
    Server says:
  </span>
   
  test-value
</div>
`;
